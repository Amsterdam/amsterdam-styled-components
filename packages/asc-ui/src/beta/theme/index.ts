import * as ascBetaTheme from './default'

export { default as ThemeProvider } from './ThemeProvider'
export { ascBetaTheme }
