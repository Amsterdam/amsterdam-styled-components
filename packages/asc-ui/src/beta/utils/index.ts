import deprecatedWarning from './deprecatedWarning'

export {
  themeColor,
  srOnlyStyle,
  breakpoint,
  svgFill,
  fluidTypoStyle,
  typographyStyle,
  showAboveBackDrop,
  getValueFromTheme,
  themeSpacing,
} from './themeUtils'
export { valueFromObject } from './valueFromObject'
export { fromProps } from './fromProps'

export { deprecatedWarning }
