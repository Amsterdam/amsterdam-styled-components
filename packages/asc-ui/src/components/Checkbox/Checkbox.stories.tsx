import React from 'react'
import { storiesOf } from '@storybook/react'
import Checkbox from './Checkbox'
import Label from '../Label'

storiesOf('Atoms/Checkbox', module)
  .addDecorator(storyFn => (
    <div
      style={{
        display: 'flex',
        maxWidth: 400,
        flexDirection: 'column',
        alignItems: 'flex-start',
        padding: '40px 10px',
      }}
    >
      {storyFn()}
    </div>
  ))
  .add('variants', () => (
    <>
      <Label htmlFor="default" label="Default">
        <Checkbox id="default" />
      </Label>
      <Label htmlFor="primary" label="Primary">
        <Checkbox id="primary" variant="primary" />
      </Label>
      <Label htmlFor="secondary" label="Secondary">
        <Checkbox id="twsecondaryo" variant="secondary" />
      </Label>
      <Label htmlFor="tertiary" label="Tertiary">
        <Checkbox id="tertiary" variant="tertiary" />
      </Label>
      <Label htmlFor="error" label="Error">
        <Checkbox id="error" error />
      </Label>
    </>
  ))
  .add('checked', () => (
    <>
      <Label htmlFor="default" label="Default checked">
        <Checkbox id="default" checked />
      </Label>
      <Label htmlFor="primary" label="Primary checked">
        <Checkbox id="primary" variant="primary" checked />
      </Label>
      <Label htmlFor="secondary" label="Secondary checked">
        <Checkbox id="secondary" variant="secondary" checked />
      </Label>
      <Label htmlFor="tertiary" label="Tertiary checked">
        <Checkbox id="tertiary" variant="tertiary" checked />
      </Label>
    </>
  ))
  .add('disabled', () => (
    <>
      <Label disabled htmlFor="disabled" label="Disabled">
        <Checkbox id="disabled" />
      </Label>
      <Label disabled htmlFor="disabledChecked" label="Disabled checked">
        <Checkbox checked id="disabledChecked" />
      </Label>
      <Label disabled htmlFor="primary" label="Disabled checked primary">
        <Checkbox checked id="primary" variant="primary" />
      </Label>
      <Label disabled htmlFor="secondary" label="Disabled checked secondary">
        <Checkbox checked id="secondary" variant="secondary" />
      </Label>
      <Label disabled htmlFor="tertiary" label="Tertiary checked secondary">
        <Checkbox checked id="tertiary" variant="tertiary" />
      </Label>
    </>
  ))
  .add('checkbox label positions', () => (
    <>
      <Label htmlFor="top" label="Top" position="top">
        <Checkbox id="top" />
      </Label>
      <Label htmlFor="right" label="Right" position="right">
        <Checkbox id="right" />
      </Label>
      <Label htmlFor="bottom" label="Bottom" position="bottom">
        <Checkbox id="bottom" />
      </Label>
      <Label htmlFor="left" label="Left" position="left">
        <Checkbox id="left" />
      </Label>
    </>
  ))
