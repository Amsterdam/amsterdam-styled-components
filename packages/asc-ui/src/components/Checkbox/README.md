# Checkbox
[Design System](https://designsystem.amsterdam.nl/7awj1hc9f/p/6495d9-checkbox)

**Component status: experimental**