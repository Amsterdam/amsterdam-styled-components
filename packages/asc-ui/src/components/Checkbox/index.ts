export { default as CheckboxStyle, CheckboxWrapperStyle } from './CheckboxStyle'
export { default as Checkbox } from './Checkbox'
