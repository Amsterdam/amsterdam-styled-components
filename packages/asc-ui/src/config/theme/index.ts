import { Theme } from '@amsterdam/asc-core'
import colors from './colors'
import typography from './typography'

export default Theme.getTheme(
  colors,
  typography,
)
