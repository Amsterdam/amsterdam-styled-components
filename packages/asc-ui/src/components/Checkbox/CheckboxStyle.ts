import styled, { css } from 'styled-components'
import { svgFill, themeColor } from '../../utils'
import { IconStyle } from '../Icon'

type CheckboxVariant = 'primary' | 'secondary' | 'tertiary'

export type Props = {
  variant?: CheckboxVariant
  checked?: boolean
  disabled?: boolean
  error?: boolean
  indeterminate?: boolean
}

const getVariant = () => ({ variant }: Props) => {
  switch (variant) {
    case 'primary':
      return css`
        color: ${themeColor('primary', 'main')};
        background-color: ${themeColor('primary', 'main')};
        ${svgFill(themeColor('tint', 'level1'))};
      `

    case 'secondary':
      return css`
        color: ${themeColor('secondary', 'main')};
        background-color: ${themeColor('secondary', 'main')};
        ${svgFill(themeColor('tint', 'level1'))};
      `

    case 'tertiary':
      return css`
        color: ${themeColor('tint', 'level7')};
        background-color: ${themeColor('tint', 'level1')};
        ${svgFill(themeColor('tint', 'level7'))};
      `

    default:
      return css`
        color: ${themeColor('tint', 'level7')};
        background-color: ${themeColor('tint', 'level7')};
        ${svgFill(themeColor('tint', 'level1'))};
      `
  }
}

export default styled.input.attrs({
  type: 'checkbox',
})<Props>`
  cursor: inherit;
  margin: 0;
  padding: 0;
  position: absolute;
  width: 24px;
  height: 24px;
  left: 6px;
  top: 6px;
`

const CheckboxIconStyle = styled(IconStyle)<Props>`
  display: flex;
  border-width: 1px;
  border-style: solid;
  width: 22px;
  height: 22px;
  position: relative;
  justify-content: center;
  align-items: center;
  z-index: 1;
  background-color: ${themeColor('tint', 'level1')};
  transition: background-color 0.2s ease-in-out;
  pointer-events: none;
  ${({ checked, indeterminate }) =>
    (checked || indeterminate) &&
    css`
      ${getVariant()};
    `};
`

const CheckboxWrapperStyle = styled.div<Props>`
  position: relative;
  display: inline-flex;
  user-select: none;
  vertical-align: middle;
  margin-bottom: 1px;
  padding: 6px;
  flex-shrink: 0; /* IE11 fix */
  color: ${themeColor('tint', 'level5')};
  ${({ disabled }) =>
    disabled &&
    css`
      pointer-events: none;
    `}
  ${({ checked, disabled }) =>
    !checked &&
    !disabled && // IE11 fix. If an element is disabled, you can still hover on it on IE11.
    css`
      &:hover {
        color: ${themeColor('tint', 'level7')};

        ${CheckboxIconStyle} {
          outline-style: solid;
          outline-width: 1px;
        }
      }
    `}
    ${({ error, checked, disabled }) =>
    error &&
    !checked &&
    !disabled &&
    css`
      ${CheckboxIconStyle} {
        border-color: red;
        outline: 1px solid red;
      }
    `}
`

export { CheckboxWrapperStyle, CheckboxIconStyle }
