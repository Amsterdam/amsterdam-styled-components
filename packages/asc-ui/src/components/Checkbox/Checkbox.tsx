import React from 'react'
import { Checkmark } from '@datapunt/asc-assets'
import CheckboxStyle, {
  CheckboxIconStyle,
  CheckboxWrapper,
  Props as StyleProps,
} from './CheckboxStyle'

type Props = {
  henk?: boolean
} & StyleProps

const Checkbox: React.FC<Props & React.HTMLAttributes<HTMLDivElement>> = ({
  checked: checkedProp,
  className,
  onChange,
  variant,
  disabled,
  ...otherProps
}) => {
  const [checked, setChecked] = React.useState(!!checkedProp)
  const [focus, setFocus] = React.useState(false)

  return (
    <CheckboxWrapper
      {...{ className, disabled, focus, checked }}
      aria-disabled={disabled}
    >
      <CheckboxIconStyle {...{ disabled, variant, checked }} size={15}>
        {checked && <Checkmark />}
      </CheckboxIconStyle>
      <CheckboxStyle
        {...{ ...otherProps, disabled, checked }}
        onFocus={() => {
          setFocus(true)
          console.log('focus set')
        }}
        onBlur={() => setFocus(false)}
        onChange={(e: React.ChangeEvent<HTMLInputElement>) => {
          if (onChange) {
            onChange(e)
          }
          setChecked(!checked)
        }}
      />
    </CheckboxWrapper>
  )
}

export default Checkbox
