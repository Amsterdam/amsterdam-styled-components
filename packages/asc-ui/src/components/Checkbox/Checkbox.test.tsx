import { createRef } from 'react'
import { render, fireEvent } from '@testing-library/react'
import Checkbox from './Checkbox'
import { ascDefaultTheme, ThemeProvider } from '../../theme'
import { themeColor } from '../../utils'

describe('Checkbox', () => {
  it('should toggle checked / not checked', () => {
    const onChangeMock = jest.fn()
    const { container } = render(
      <ThemeProvider>
        <Checkbox variant="primary" onChange={onChangeMock} />
      </ThemeProvider>,
    )

    const checkBox = container.querySelector('input')
    const icon = container.querySelector('span')

    expect(icon).toHaveStyleRule(
      'background-color',
      themeColor('tint', 'level1')({ theme: ascDefaultTheme }),
    )

    // Toggle on
    if (checkBox) {
      fireEvent.click(checkBox)
    }

    expect(onChangeMock).toHaveBeenCalled()
    expect(icon).not.toHaveStyleRule(
      'background-color',
      themeColor('tint', 'level1')({ theme: ascDefaultTheme }),
    )

    // Toggle off
    if (checkBox) {
      fireEvent.click(checkBox)
    }

    expect(icon).toHaveStyleRule(
      'background-color',
      themeColor('tint', 'level1')({ theme: ascDefaultTheme }),
    )
  })

  it('should toggle checked / not checked when the props change', () => {
    const onChangeMock = jest.fn()
    const { container, rerender } = render(
      <ThemeProvider>
        <Checkbox variant="primary" onChange={onChangeMock} />
      </ThemeProvider>,
    )

    const icon = container.querySelector('span')

    expect(icon).toHaveStyleRule(
      'background-color',
      themeColor('tint', 'level1')({ theme: ascDefaultTheme }),
    )

    // Toggle on by changing props
    rerender(
      <ThemeProvider>
        <Checkbox checked variant="secondary" />
      </ThemeProvider>,
    )

    expect(icon).not.toHaveStyleRule(
      'background-color',
      themeColor('tint', 'level1')({ theme: ascDefaultTheme }),
    )

    // Toggle of by changing props
    rerender(
      <ThemeProvider>
        <Checkbox />
      </ThemeProvider>,
    )

    expect(icon).toHaveStyleRule(
      'background-color',
      themeColor('tint', 'level1')({ theme: ascDefaultTheme }),
    )
  })

  it('should handle refs', () => {
    const ref = createRef<HTMLInputElement>()

    render(
      <ThemeProvider>
        <Checkbox ref={ref} />
      </ThemeProvider>,
    )

    expect(ref.current).toBeInstanceOf(HTMLInputElement)
    expect(ref.current?.type).toEqual('checkbox')
  })
})
