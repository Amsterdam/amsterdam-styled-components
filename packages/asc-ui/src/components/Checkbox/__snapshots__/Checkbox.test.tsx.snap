// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should render 1`] = `
.c3 {
  top: 0;
  left: 0;
  width: 100%;
  cursor: inherit;
  height: 100%;
  margin: 0;
  opacity: 0;
  padding: 0;
  position: absolute;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 15px;
  height: 15px;
  box-sizing: content-box;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border-width: 1px;
  border-style: solid;
  width: 22px;
  height: 22px;
  position: relative;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition: background-color 0.2s ease-in-out;
  transition: background-color 0.2s ease-in-out;
}

.c2 > svg {
  width: 15px;
  height: 15px;
}

.c0 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  margin-bottom: 1px;
  padding: 6px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  color: #767676;
}

.c0:hover {
  color: #000000;
}

.c0:hover .c1 {
  outline-style: solid;
  outline-width: 1px;
}

.c0.c0:focus {
  outline-color: #FEC813;
  outline-style: solid;
  outline-offset: 0px;
  outline-width: 3px;
}

<div>
  <div
    class="c0"
  >
    <span
      class="c1 c2"
      size="15"
    />
    <input
      class="c3"
      type="checkbox"
    />
  </div>
</div>
`;
